llc/llc-evolution.md

```diff
@@ -87,11 +87,11 @@ public virtual Response GetPetsByName(string petName, string sort, int? limit, i
 public virtual Response GetPetsByName(string petName, RequestParameters parameters, RequestOptions requestOptions = default);
 ```
 
-This design is predicated on a new `RequestParameters` type, added to Azure Core, which is a property bag mapping string names to values. Logically, this type is similar to `Dictionary<string, object>` but it optimized for holding a small number of key value pairs.
+This design is predicated on a new `RequestParameters` type, added to Azure Core, which is a property bag mapping string names to values. Logically, this type is similar to `Dictionary<string, object>` but is optimized for holding a small number of key value pairs.
 
 Note that the `RequestParameters` type is not specific to a specific operation, which means it will not have members that correspond to parameters for the operation and so developers will have to depend on documentation to understand the properties the service supports and their names as well as their types.
 
-*(N.B: We have not closed on which of the two strategies we will employ, and the `RequestParameters` strategy would require some work in Azure Core, fortunately, we can start with the first strategy of just adding overloads with more and more optional parameters and then cut over to the new plan if we are comfortable with it)*
+*(N.B: We have not closed on which of the two strategies we will employ, and the `RequestParameters` strategy would require some work in Azure Core, fortunately, we can start with the first strategy of just adding overloads with more and more optional parameters and then cut over to the new plan if we are comfortable with it.)*
```

Comment:
Is this still an open question?  I hadn't heard yet about `RequestParameters`, but maybe we're holding it in a back pocket given the ability to move to it later.

Reply:
I think the question is still open - the current strategy of adding overloads will work, it will just create a bunch of method overloads.  I think there will be a tension here between the following choices:

1. Have formal method parameters for every operation parameter. (Works, but can lead to very long method signatures, especially for methods which have many optional parameters).
2. Operation specific Options classes (e.g. NewPetOptions) which has a property for each option (method signatures are nicer, but we end up now (possibly) generating a type per operation to hold the options.  I do think in practice this may not be the end of the world, especially if it was just optional parameters.)
3. A middle ground between 1 and 2 where instead of creating a specific type we use some general RequestParameters thing.  This will mean fewer types in the system, but the ergonomics will be worse (similar to how by not generating model classes we make the ergonomics for request bodies worse).